cmd/builder/internal/builder/main.go

```diff
@@ -110,16 +111,18 @@ func Compile(cfg Config) error {
 
 	cfg.Logger.Info("Compiling")
 
-	var ldflags = "-s -w"
-
+	ldflags := cfg.LDFlags
 	args := []string{"build", "-trimpath", "-o", cfg.Distribution.Name}
 	if cfg.Distribution.DebugCompilation {
 		cfg.Logger.Info("Debug compilation is enabled, the debug symbols will be left on the resulting binary")
-		ldflags = cfg.LDFlags
 		args = append(args, "-gcflags=all=-N -l")
-	} else if len(cfg.LDFlags) > 0 {
-		ldflags += " " + cfg.LDFlags
+
+		if cfg.LDFlags == DefaultLDFlags {
+			// debug_compilation enabled should not strip native debug and DWARF symbols
+			ldflags = ""
```

Comment:
I don't understand this logic, sorry.

Reply:
this is the part where `debug_compilation: true` so IMO, if `--ldflags` arg is not changed, it should not strip debug and DWARF symbols